File: javascript/filters/css.js
/**
 * Process CSS properties: replaces snippets, augumented with ! char, with 
 * <em>!important</em> suffix 
 * @author Sergey Chikuyonok (serge.che@gmail.com)
 * @link http://chikuyonok.ru
 */(function(){
	var re_important = /(.+)\!$/;
	function process(tree, profile) {
		for (var i = 0, il = tree.children.length; i < il; i++) {
			/** @type {ZenNode} */
			var item = tree.children[i];
			
			// CSS properties are always snippets
			if (item.type == 'snippet' && re_important.test(item.real_name)) {
				item.start = item.start.replace(/(;?)$/, ' !important$1');
			}
			
			process(item, profile);
		}
		
		return tree;
	}
	
	zen_coding.registerFilter('css', process);
})();

File: python/zencoding/filters/css.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
'''
Process CSS properties: replaces snippets, augumented with ! char, with 
<em>!important</em> suffix
@author Sergey Chikuyonok (serge.che@gmail.com)
@link http://chikuyonok.ru
'''
import re
import zencoding

re_important = re.compile(r'(.+)\!$')

@zencoding.filter('css')
def process(tree, profile):
	for item in tree.children:
		# CSS properties are always snippets
		if item.type == 'snippet' and re_important.search(item.real_name):
			item.start = re.sub(r'(;?)$', ' !important\\1', item.start)
		
		process(item, profile)
	
	return tree